packages/wrangler/src/__tests__/middleware.scheduled.test.ts

```diff
@@ -0,0 +1,135 @@
+import * as fs from "node:fs";
+import { unstable_dev } from "../api";
+import { runInTempDir } from "./helpers/run-in-tmp";
+
+jest.unmock("undici");
+
+describe("run scheduled events with middleware", () => {
```

Comment:
I was a little worried when I started the whole middleware thing because there wasn't a particularly nice way of testing any of this, but this has made it super easy